File: 004_valid-anagrams/Valid_Anagrams.py
def valid_anagrams(s, t):
	if len(s) != len(t): return False

	chars = [0] * 26

	for i in range(len(s)):
		chars[ord(s[i]) - ord("a")] += 1
		chars[ord(t[i]) - ord("a")] -= 1

	for char in chars:
		if char != 0: return False

	return True

def main():
	s = "cat"
	t = "tac"
	print("Is Anagram ? ", valid_anagrams(s = "garage", t = "ragage"))


if __name__ == "__main__":
	main()

File: 004_valid-anagrams/ValidAnagrams.java
public class ValidAnagrams {
	public static boolean validAnagrams(String s, String t) {
		if(s.length() != t.length()) return false;

		int[] chars = new int[26];

		for(int i = 0; i< s.length(); i++) {
			chars[s.charAt(i) - 'a']++;
			chars[t.charAt(i) - 'a']--;
		}

		for(int ch: chars) {
			if(ch != 0) return false;
		}

		return true;
	}
	public static void main(String[] args) {
		String s = "garage";
		String t = "ragega";

		System.out.println("Is Anagram ? " + validAnagrams(s, t));
	}
}
